components/Competition/Banner/index.vue

```diff
@@ -36,6 +36,12 @@ export default {
   methods: {
     onClickCTA () {
       this.$emit('onClickCTA', true)
+    },
+    showQuestionnaire () {
+      /**
+       * Trigger to open questionnaire page.
+       */
+      this.$router.push({ path: '/registration', query: { option: 'desa' } })
```

Comment:
in my opinion, we can just leave the function to emit onClickCTA on each components section, and make the reusable function in the competition index, but this is ok 👌

Reply:
It's a nice solution, Thanks ⭐ . I have another approach to creating this.

First, in `pages/sayembara.vue` I can add conditional. See picture below.
![showModal](https://user-images.githubusercontent.com/79241754/178638565-8bc8453e-501c-4dbb-95a8-31ce5f341005.png)

Then, in `components/Competition/Join/index.vue` on the `onClickCTA` method I pass the second argument to provide the value `true` to show modal and `false` to open questionnaire page
![onClickCTA](https://user-images.githubusercontent.com/79241754/178639049-9a1867ee-b8eb-4698-84d0-e39141bb5c8a.png)

And what do you think❓ 
@yoslie

Reply:
Sure this is ok, but from the function name we can see from the reading side showModal means to open modal, but questionnaire is a page.. 
so maybe either renaming the function (make it more global) or make a new function is the solution #cmiiw

Reply:
Yes. it's quite confusing to rename the function to open the modal and the questionnaire page. I think it's good to create a new method to open the questionnaire page in `pages/sayembara.vue`. Okay, thanks for the discussion 🌟